src/test/java/BaseTest.java

```diff
@@ -1,11 +1,69 @@
 import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
 import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Parameters;
+
+import java.time.Duration;
 
 public class BaseTest {
+    public WebDriver driver;
+    public WebDriverWait wait;
+    public Actions actions;
+    public String url = "https://qa.koel.app/";
 
     @BeforeSuite
     static void setupClass() {
         WebDriverManager.chromedriver().setup();
     }
-}
\ No newline at end of file
+
+    @BeforeMethod
+    @Parameters({"BaseURL"})
+    public void launchBrowser(String baseURL){
+        ChromeOptions options = new ChromeOptions();
+        options.addArguments("--remote-allow-origins=*");
+        driver = new ChromeDriver(options);
+        //Implicit Wait
+        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
+        driver.manage().window().maximize();
+        //Explicit Wait
+        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+        driver.manage().window().maximize();
+        actions = new Actions(driver);
+        navigateToPage();
```

Comment:
Adding multiple arguments:
`--disable-notifications`: disabling notifications
`--remote-allow-origins=*`: allowing remote origins
`--incognito`: launching in incognito mode
`--start-maximized`: launching in maximized mode
`"excludeSwitches", new String[]{"enable-automation"} `: Disable notification Chrome is being controlled by automated test software in ChromeDriver
**These settings are enabled to simplify and improve the process of automated testing.**
_For example_
![image](https://github.com/Test-Pro-Bootcamp/A57-QA-Automation-Framework/assets/104761482/b05c1900-365a-491c-a399-92b5b3bdc666)
```
                WebDriverManager.chromedriver().setup();
                ChromeOptions options = new ChromeOptions();
                options.addArguments("--disable-notifications","--remote-allow-origins=*", "--incognito","--start-maximized");
                options.setExperimentalOption("excludeSwitches", new String[]{"enable-automation"});
                return new ChromeDriver(options);

```